thewalrus/tests/test_hafnian.py

```diff
@@ -277,3 +277,12 @@ def test_diag(self, n):
         haf = hafnian(A, loop=True)
         expected = np.prod(v)
         assert np.allclose(haf, expected)
+
+
+@pytest.mark.parametrize("n", [2, 3, 5, 10, 15, 20])
+@pytest.mark.parametrize("fill", [0.5, 0.2, 0.1, 0.05])
+def test_valid_output(random_matrix, n, fill):
```

Comment:
How come this test was not included in the test class? Could add it by indenting it and adding `self` as the first argument.

Reply:
done